File: Data Structures & Algorithms/longest-common-prefix/submission-1.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs or not strs[0]:
            return ""
        first = strs[0]
        for i in range(len(strs[0])):
            char = first[i]
            for s in strs[1:]:
                if i >= len(s) or s[i] != char:
                    return first[:i]
        return first

File: Data Structures & Algorithms/string-encode-and-decode/submission-1.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string encoded="";
        for(string x:strs){
            encoded+=to_string(x.length())+"#"+x;
        }
        return encoded;
    }

    vector<string> decode(string s) {
        vector<string> result;
        int i = 0;
        while (i<s.length()){
            int j=i;
            while(s[j]!='#'){
                j++;
            }
            //get length
            int len= stoi(s.substr(i,j-i));
            j++;
            string str=s.substr(j,len);
            result.push_back(str);
            i=j+len;
        }
        return result;
    }
};
